### ISSUE TITLE
Unable to run Lean 4 example from docs

### ISSUE THREAD

**Issue by AG161** [lean-dojo/LeanDojo#3]
From "Getting Started" in the docs: https://leandojo.readthedocs.io/en/latest/getting-started.html

I ran the following on MacOS 13.4.1:

```python
from lean_dojo import LeanGitRepo, trace

repo = LeanGitRepo("https://github.com/yangky11/lean4-example", "805b3ad26fd1e6c8ba507f41ab289ec19c0efe2c")
trace(repo, dst_dir="traced_lean4-example")
``` 

Here is the result (VERBOSE = 1):

```shell
(atp) agittis@Andreass-MBP Dojo % sw_vers
ProductName:		macOS
ProductVersion:		13.4.1
BuildVersion:		22F82
(atp) agittis@Andreass-MBP Dojo % python
Python 3.9.6 (default, May  7 2023, 23:32:44) 
[Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from lean_dojo import LeanGitRepo, trace
>>> repo = LeanGitRepo("https://github.com/yangky11/lean4-example", "805b3ad26fd1e6c8ba507f41ab289ec19c0efe2c")
>>> trace(repo, dst_dir="traced_lean4-example")
2023-07-03 17:30:36.916 | DEBUG    | lean_dojo.data_extraction.trace:get_traced_repo_path:140 - The traced repo is available in the cache.
2023-07-03 17:30:36.917 | INFO     | lean_dojo.data_extraction.trace:trace:164 - Loading the traced repo from /Users/agittis/.cache/lean_dojo/yangky11-lean4-example-805b3ad26fd1e6c8ba507f41ab289ec19c0efe2c/lean4-example
2023-07-03 17:30:37.047 | DEBUG    | lean_dojo.data_extraction.traced_data:load_from_disk:1273 - Loading 620 traced XML files from /Users/agittis/.cache/lean_dojo/yangky11-lean4-example-805b3ad26fd1e6c8ba507f41ab289ec19c0efe2c/lean4-example with 7 workers
2023-07-03 17:30:38,764	INFO worker.py:1627 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8265 
 15%|██████▍                                   | 95/620 [00:08<00:33, 15.47it/s]2023-07-03 17:30:47,549	WARNING worker.py:2019 -- Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 1431, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1510, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1434, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1438, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1378, in ray._raylet.execute_task.function_executor
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/function_manager.py", line 724, in actor_method_executor
    return method(__ray_actor, *args, **kwargs)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/util/tracing/tracing_helper.py", line 464, in _resume_span
    return method(self, *_args, **_kwargs)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/lean_dojo/data_extraction/traced_data.py", line 1080, in load_xml_from_disk
    return TracedFile.from_xml(self.root_dir, path, self.repo)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/lean_dojo/data_extraction/traced_data.py", line 1022, in from_xml
    tree = etree.parse(path).getroot()
  File "src/lxml/etree.pyx", line 3541, in lxml.etree.parse
  File "src/lxml/parser.pxi", line 1879, in lxml.etree._parseDocument
  File "src/lxml/parser.pxi", line 1905, in lxml.etree._parseDocumentFromURL
  File "src/lxml/parser.pxi", line 1808, in lxml.etree._parseDocFromFile
  File "src/lxml/parser.pxi", line 1180, in lxml.etree._BaseParser._parseDocFromFile
  File "src/lxml/parser.pxi", line 618, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 728, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 657, in lxml.etree._raiseParseError
  File "/Users/agittis/.cache/lean_dojo/yangky11-lean4-example-805b3ad26fd1e6c8ba507f41ab289ec19c0efe2c/lean4-example/lake-packages/lean4/lib/lean/Init/Control/Except.trace.xml", line 2828
lxml.etree.XMLSyntaxError: Premature end of data in tag CommandDeclvalsimpleNode4 line 2819, line 2828, column 10

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 1796, in ray._raylet.task_execution_handler
  File "python/ray/_raylet.pyx", line 1695, in ray._raylet.execute_task_with_cancellation_handler
  File "python/ray/_raylet.pyx", line 1380, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1595, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 875, in ray._raylet.store_task_errors
  File "python/ray/_raylet.pyx", line 3456, in ray._raylet.CoreWorker.store_task_outputs
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/serialization.py", line 466, in serialize
    return self._serialize_to_msgpack(value)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/serialization.py", line 421, in _serialize_to_msgpack
    value = value.to_bytes()
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/exceptions.py", line 32, in to_bytes
    serialized_exception=pickle.dumps(self),
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 88, in dumps
    cp.dump(obj)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 733, in dump
    return Pickler.dump(self, obj)
TypeError: cannot pickle 'lxml.etree._ListErrorLog' object
An unexpected internal error occurred while the worker was executing a task.
2023-07-03 17:30:47,576	WARNING worker.py:2019 -- A worker died or was killed while executing a task by an unexpected system error. To troubleshoot the problem, check the logs for the dead worker. RayTask ID: ffffffffffffffff8ec4d6f2cd6540394f4af10201000000 Worker ID: 608ac959092109d92921d0db0824029b71e2354ce10d7771b600ad7e Node ID: 06c5d34f6c091588f48067732fdc951ea28695ca70aa80c4f6d73e15 Worker IP address: 127.0.0.1 Worker port: 52804 Worker PID: 2102 Worker exit type: SYSTEM_ERROR Worker exit detail: Worker exits unexpectedly. Worker exits with an exit code None.
 Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 1431, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1510, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1434, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1438, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1378, in ray._raylet.execute_task.function_executor
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/function_manager.py", line 724, in actor_method_executor
    return method(__ray_actor, *args, **kwargs)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/util/tracing/tracing_helper.py", line 464, in _resume_span
    return method(self, *_args, **_kwargs)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/lean_dojo/data_extraction/traced_data.py", line 1080, in load_xml_from_disk
    return TracedFile.from_xml(self.root_dir, path, self.repo)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/lean_dojo/data_extraction/traced_data.py", line 1022, in from_xml
    tree = etree.parse(path).getroot()
  File "src/lxml/etree.pyx", line 3541, in lxml.etree.parse
  File "src/lxml/parser.pxi", line 1879, in lxml.etree._parseDocument
  File "src/lxml/parser.pxi", line 1905, in lxml.etree._parseDocumentFromURL
  File "src/lxml/parser.pxi", line 1808, in lxml.etree._parseDocFromFile
  File "src/lxml/parser.pxi", line 1180, in lxml.etree._BaseParser._parseDocFromFile
  File "src/lxml/parser.pxi", line 618, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 728, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 657, in lxml.etree._raiseParseError
  File "/Users/agittis/.cache/lean_dojo/yangky11-lean4-example-805b3ad26fd1e6c8ba507f41ab289ec19c0efe2c/lean4-example/lake-packages/lean4/lib/lean/Init/Control/Except.trace.xml", line 2828
lxml.etree.XMLSyntaxError: Premature end of data in tag CommandDeclvalsimpleNode4 line 2819, line 2828, column 10

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 1796, in ray._raylet.task_execution_handler
  File "python/ray/_raylet.pyx", line 1695, in ray._raylet.execute_task_with_cancellation_handler
  File "python/ray/_raylet.pyx", line 1380, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1595, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 875, in ray._raylet.store_task_errors
  File "python/ray/_raylet.pyx", line 3456, in ray._raylet.CoreWorker.store_task_outputs
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/serialization.py", line 466, in serialize
    return self._serialize_to_msgpack(value)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/serialization.py", line 421, in _serialize_to_msgpack
    value = value.to_bytes()
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/exceptions.py", line 32, in to_bytes
    serialized_exception=pickle.dumps(self),
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 88, in dumps
    cp.dump(obj)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 733, in dump
    return Pickler.dump(self, obj)
TypeError: cannot pickle 'lxml.etree._ListErrorLog' object
An unexpected internal error occurred while the worker was executing a task.
 16%|██████▋                                   | 98/620 [00:08<00:44, 11.86it/s]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/lean_dojo/data_extraction/trace.py", line 165, in trace
    traced_repo = TracedRepo.load_from_disk(cached_path)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/lean_dojo/data_extraction/traced_data.py", line 1283, in load_from_disk
    traced_files = list(
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/tqdm/std.py", line 1178, in __iter__
    for obj in iterable:
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/util/actor_pool.py", line 126, in get_generator
    yield self.get_next_unordered()
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/util/actor_pool.py", line 273, in get_next_unordered
    return ray.get(future)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/auto_init_hook.py", line 18, in auto_init_wrapper
    return fn(*args, **kwargs)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/client_mode_hook.py", line 103, in wrapper
    return func(*args, **kwargs)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/worker.py", line 2542, in get
    raise value
ray.exceptions.RayActorError: The actor died unexpectedly before finishing this task.
	class_name: _TracedRepoHelper
	actor_id: 8ec4d6f2cd6540394f4af10201000000
	pid: 2102
	namespace: 19c86c3f-d938-4203-ad25-de0d3b994371
	ip: 127.0.0.1
The actor is dead because its worker process has died. Worker exit type: SYSTEM_ERROR Worker exit detail: Worker exits unexpectedly. Worker exits with an exit code None.
 Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 1431, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1510, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1434, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1438, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1378, in ray._raylet.execute_task.function_executor
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/function_manager.py", line 724, in actor_method_executor
    return method(__ray_actor, *args, **kwargs)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/util/tracing/tracing_helper.py", line 464, in _resume_span
    return method(self, *_args, **_kwargs)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/lean_dojo/data_extraction/traced_data.py", line 1080, in load_xml_from_disk
    return TracedFile.from_xml(self.root_dir, path, self.repo)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/lean_dojo/data_extraction/traced_data.py", line 1022, in from_xml
    tree = etree.parse(path).getroot()
  File "src/lxml/etree.pyx", line 3541, in lxml.etree.parse
  File "src/lxml/parser.pxi", line 1879, in lxml.etree._parseDocument
  File "src/lxml/parser.pxi", line 1905, in lxml.etree._parseDocumentFromURL
  File "src/lxml/parser.pxi", line 1808, in lxml.etree._parseDocFromFile
  File "src/lxml/parser.pxi", line 1180, in lxml.etree._BaseParser._parseDocFromFile
  File "src/lxml/parser.pxi", line 618, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 728, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 657, in lxml.etree._raiseParseError
  File "/Users/agittis/.cache/lean_dojo/yangky11-lean4-example-805b3ad26fd1e6c8ba507f41ab289ec19c0efe2c/lean4-example/lake-packages/lean4/lib/lean/Init/Control/Except.trace.xml", line 2828
lxml.etree.XMLSyntaxError: Premature end of data in tag CommandDeclvalsimpleNode4 line 2819, line 2828, column 10

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 1796, in ray._raylet.task_execution_handler
  File "python/ray/_raylet.pyx", line 1695, in ray._raylet.execute_task_with_cancellation_handler
  File "python/ray/_raylet.pyx", line 1380, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1595, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 875, in ray._raylet.store_task_errors
  File "python/ray/_raylet.pyx", line 3456, in ray._raylet.CoreWorker.store_task_outputs
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/serialization.py", line 466, in serialize
    return self._serialize_to_msgpack(value)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/_private/serialization.py", line 421, in _serialize_to_msgpack
    value = value.to_bytes()
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/exceptions.py", line 32, in to_bytes
    serialized_exception=pickle.dumps(self),
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 88, in dumps
    cp.dump(obj)
  File "/Users/agittis/ProofSim/atp/lib/python3.9/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 733, in dump
    return Pickler.dump(self, obj)
TypeError: cannot pickle 'lxml.etree._ListErrorLog' object
An unexpected internal error occurred while the worker was executing a task.
>>> 
``` 

I managed to run the Lean 3 example from the same page with no issue. Sorry in advance if this is an issue on my end, I have very little technical expertise. Cheers.

**Comment by yangky11** [comment]
Hi @AG161,

It looks like the error is due to LeanDojo trying to read from a corrupted cache. Could you please delete `~/.cache/lean_dojo` and re-try? Thank you

**Comment by yangky11** [comment]
I have also updated the example, please use the commit `7d711f6da4584ecb7d4f057715e1f72ba175c910` instead. It should be faster than `805b3ad26fd1e6c8ba507f41ab289ec19c0efe2c ` due to the removal of unnecessary mathlib dependency.

**Comment by AG161** [comment]
Hi @yangky11,

I think that fixes it. I did delete the cache once when I was troubleshooting but maybe it got corrupted again. Using the commit you linked the code worked the second time I ran it. The first time I ran it, it got stuck for a few hours on the attached imaged. I interrupted it and then when I ran it again it worked. Thanks!
<img width="809" alt="Screenshot 2023-07-03 at 8 32 38 PM" src="https://github.com/lean-dojo/LeanDojo/assets/81839540/7d37a75f-4608-447e-8d52-d283131b5c56">
